exchanges/okx/okx.go

```diff
@@ -0,0 +1,5085 @@
+	errIncompleteCandlestickData                     = errors.New("incomplete candlestick data")
+	errMaxRFQOrdersToCancel                          = errors.New("no more than 100 RFQ cancel order parameter is allowed")
+	errMalformedData                                 = errors.New("malformed data")
+	errInvalidUnderlying                             = errors.New("invalid underlying")
+	errMissingRequiredParameter                      = errors.New("missing required parameter")
+	errMissingMakerInstrumentSettings                = errors.New("missing maker instrument settings")
+	errInvalidSubAccountName                         = errors.New("invalid sub-account name")
+	errInvalidAPIKey                                 = errors.New("invalid api key")
+	errInvalidAlgoID                                 = errors.New("invalid algo id")
+	errInvalidMarginTypeAdjust                       = errors.New("invalid margin type adjust, only 'add' and 'reduce' are allowed")
+	errInvalidAlgoOrderType                          = errors.New("invalid algo order type")
+	errEmptyArgument                                 = errors.New("empty argument")
+	errInvalidCurrencyPair                           = errors.New("invalid currency pair")
+	errCurrencyPairNotAvailable                      = errors.New("currency pair not available")
+)
+
+/************************************ MarketData Endpoints *************************************************/
+
+// PlaceOrder place an order only if you have sufficient funds.
+func (ok *Okx) PlaceOrder(ctx context.Context, arg *PlaceOrderRequestParam) (*PlaceOrderResponse, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	if !(arg.TradeMode == TradeModeCross || arg.TradeMode == TradeModeIsolated || arg.TradeMode == TradeModeCash) {
+		return nil, errInvalidTradeModeValue
+	}
+	arg.Side = strings.ToUpper(arg.Side)
+	if !(arg.Side == order.Buy.String() || arg.Side == order.Sell.String()) {
+		return nil, errMissingOrderSide
+	}
+	arg.OrderType = strings.ToUpper(arg.OrderType)
+	if !(arg.OrderType == OkxOrderMarket ||
+		arg.OrderType == OkxOrderLimit ||
+		arg.OrderType == OkxOrderPostOnly ||
+		arg.OrderType == OkxOrderFOK ||
+		arg.OrderType == OkxOrderIOC ||
+		arg.OrderType == OkxOrderOptimalLimitIOC) {
+		return nil, errInvalidOrderType
+	}
+	if arg.QuantityToBuyOrSell <= 0 {
+		return nil, errInvalidQuantityToButOrSell
+	}
+	if arg.OrderPrice <= 0 && (arg.OrderType == OkxOrderLimit ||
+		arg.OrderType == OkxOrderPostOnly ||
+		arg.OrderType == OkxOrderFOK ||
+		arg.OrderType == OkxOrderIOC) {
+		return nil, fmt.Errorf("invalid order price for %s order types", arg.OrderType)
+	}
+	if !(arg.QuantityType == "base_ccy" ||
+		arg.QuantityType == "quote_ccy") {
+		arg.QuantityType = ""
+	}
+	type response struct {
+		Msg  string               `json:"msg"`
```

Comment:
This is because even with http Status Code 200 the data may not have whats required in the Data ( 'data' ) field.
So, checking the field 'msg' is required.